iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -279,6 +369,29 @@ public static Anonymizer write(Network n, Path xmlFile) {
         return write(n, new ExportOptions(), xmlFile);
     }
 
+    public static void write(Network network, ExportOptions options, DataSource dataSource) throws IOException {
+        Anonymizer anonymizer;
+        try (OutputStream osb = dataSource.newOutputStream(null, XIIDM, false);
+             BufferedOutputStream bos = new BufferedOutputStream(osb)) {
+            if (options.isSeparateBaseAndExtensions() && !getNetworkExtensions(network).isEmpty() && !options.isOneFilePerExtensionType()) {
+                try (OutputStream ose = dataSource.newOutputStream("ext", XIIDM, false);
+                     BufferedOutputStream bose = new BufferedOutputStream(ose)) {
+                    anonymizer = NetworkXml.write(network, options, bos, bose);
+                }
+            } else if (options.isOneFilePerExtensionType() && !getNetworkExtensions(network).isEmpty()) {
+                anonymizer = NetworkXml.write(network, options, bos, dataSource);
```

Comment:
I think this can be refactored, you do similar things in these two cases. Is it possible ?